erpnext/setup/doctype/sales_person/sales_person.js

```diff
@@ -10,6 +10,15 @@ frappe.ui.form.on('Sales Person', {
 		}
 	},
 
+	employee: function(frm) {
+		if (frm.doc.employee)
+			frm.set_df_property('sales_person_name', 'read_only', 1);
+		else{
+			frm.set_df_property('sales_person_name', 'read_only', 0);
+			frappe.model.set_value(frm.doctype, frm.docname, 'sales_person_name', "");
+		}
```

Comment:
Shouldn't the sales person name reset happen automatically with the fetch defined in the JSON?

```suggestion
		frm.toggle_enable("sales_person_name", !!frm.doc.employee);
		if (!frm.doc.employee) {
			frm.set_value("sales_person_name", "");
		}
```

Reply:
@nmimsnikhil, I tested this out. I've tweaked the suggestion, check it out.